src/solvers/flattening/bv_pointers.cpp

```diff
@@ -462,14 +462,14 @@ bvt bv_pointerst::convert_bitvector(const exprt &expr)
   if(expr.type().id()==ID_pointer)
     return convert_pointer_type(expr);
 
-  if(expr.id()==ID_minus &&
-     expr.operands().size()==2 &&
-     expr.op0().type().id()==ID_pointer &&
-     expr.op1().type().id()==ID_pointer)
+  if(
+    expr.id() == ID_minus &&
```

Comment:
This I think really cries out for using the cast style --
```
const minus_exprt *minus_expr
if((minus_expr = expr_try_dynamic_cast<minus_exprt>(expr)) &&
   minus_expr.op0()... &&
   minus_expr.op1()...
```

Reply:
Or lift that into a helper `is_pointer_subtraction`

Reply:
done